Modules/_winapi.c

```diff
@@ -464,6 +474,41 @@ _winapi_CreateFile_impl(PyObject *module, LPCTSTR file_name,
     return handle;
 }
 
+/*[clinic input]
+_winapi.CreateFileMapping -> HANDLE
+
+    file_handle: HANDLE
+    security_attributes: LPSECURITY_ATTRIBUTES
+    protect: DWORD
+    max_size_high: DWORD
+    max_size_low: DWORD
+    name: LPCWSTR
+    /
+[clinic start generated code]*/
+
+static HANDLE
+_winapi_CreateFileMapping_impl(PyObject *module, HANDLE file_handle,
+                               LPSECURITY_ATTRIBUTES security_attributes,
+                               DWORD protect, DWORD max_size_high,
+                               DWORD max_size_low, LPCWSTR name)
+/*[clinic end generated code: output=6c0a4d5cf7f6fcc6 input=3dc5cf762a74dee8]*/
+{
+    HANDLE handle;
+
+    Py_BEGIN_ALLOW_THREADS
+    handle = CreateFileMappingW(file_handle, security_attributes,
+                                protect, max_size_high, max_size_low,
+                                name);
+    Py_END_ALLOW_THREADS
+
+    if (handle == NULL) {
+        PyErr_SetFromWindowsErrWithUnicodeFilename(0, name);
+        handle = INVALID_HANDLE_VALUE;
+    }
+
+    return handle;
```

Comment:
Shouldn't you `return NULL` instead of `INVALID_HANDLE_VALUE` in case of error? It appears this way you'll return an int instead of causing an exception.

Reply:
[`HANDLE_return_converter`](https://github.com/python/cpython/blob/caf0a5d9cdfd569c438ce85ece952eacdf4100b8/Modules/_winapi.c#L171) checks for `INVALID_HANDLE_VALUE` with an error set. For `NULL` it returns `None`, which supports `GetStdHandle` since a standard-handle value may be `NULL`. Also, since an error must also be set, `GetCurrentProcess` works with the same converter even though the value it returns is equivalent to `INVALID_HANDLE_VALUE`.

It's  a bit confusing if you're not aware of how it's implemented. It would probably be better to have a special handle type for functions that fail with `INVALID_HANDLE_VALUE`, since by far the majority of WINAPI functions that return a handle (directly or as an out parameter) either use `NULL` or `FALSE` to indicate an error.

Reply:
@eryksun: Would it make more sense to handle the change you're suggesting by opening a separate issue and PR for it?

I'm hesitant to make such a "code maintenance" change impacting unrelated code as part of this PR which is an "enhancement".

@eryksun: I will offer to help supply the patch if you will review it.

Reply:
It's a minor change, but @zooba should approve it in a new issue before going forward with a PR. I'd rename the current type to `HANDLE_I` and add a new `HANDLE` type that uses `NULL` for errors.

Another maintenance issue is removing the use of the `TCHAR` API and unintentional linking with ANSI functions such as `CreateFileA`, which I think would be fine to fix here if you want. The internal use of the affected functions hasn't had a problem thus far because they use ASCII strings, e.g. for pipe names, but it may eventually cause an issue.

Reply:
Let's capture all of that in a new issue on b.p.o.